baselines/flwr_baselines/models/leaf/shakespeare.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import torch
+import torch.nn as nn
+
+from flwr.baselines.dataloaders.leaf.shakespeare import LEAF_CHARACTERS
+
+
+class ShakespeareLeafNet(nn.Module):
+    def __init__(
+        self,
+        chars: str = LEAF_CHARACTERS,
+        seq_len: int = 80,
+        hidden_size: int = 256,
+        embedding_dim=8,
+    ):
```

Comment:
need to pass the number of classes (i.e. number of potential next characters)

Reply:
https://github.com/TalwalkarLab/leaf/blob/master/models/shakespeare/stacked_lstm.py